File: amethyst/Configurations/ControlPanelViewConfiguration.cs
namespace amethyst.Configurations;

public record ControlPanelViewConfiguration(
    bool ShowClockControls,
    bool ShowScoreControls,
    bool ShowStatsControls,
    bool ShowLineupControls,
    bool ShowClocks,
    bool ShowTimeoutList,
    bool ShowScoreSheet,
    bool ShowTimeline,
    DisplaySide DisplaySide
);

public enum DisplaySide
{
    Home,
    Away,
    Both,
}

public class ControlPanelViewConfigurationFactory : IConfigurationFactory<ControlPanelViewConfiguration>
{
    public ControlPanelViewConfiguration GetDefaultValue() => new(
        true,
        true,
        true,
        true,
        true,
        false,
        true,
        false,
        DisplaySide.Both);
}

File: jamster.ui/src/types/ControlPanelViewConfiguration.ts
import { DisplaySide } from "./DisplaySide";

export type ControlPanelViewConfiguration = {
    showClockControls: boolean;
    showScoreControls: boolean;
    showStatsControls: boolean;
    showLineupControls: boolean;
    showClocks: boolean;
    showTimeoutList: boolean;
    showScoreSheet: boolean;
    showTimeline: boolean;
    displaySide: DisplaySide;
}

export const DEFAULT_CONTROL_PANEL_VIEW_CONFIGURATION: ControlPanelViewConfiguration = {
    showClockControls: true,
    showScoreControls: true,
    showStatsControls: true,
    showLineupControls: true,
    showClocks: true,
    showTimeoutList: true,
    showScoreSheet: true,
    showTimeline: false,
    displaySide: DisplaySide.Both,
}